sockeye/inference.py

```diff
@@ -379,35 +399,46 @@ def load_models(context: mx.context.Context,
                                         vocabulary.  Used when logits/softmax are handled separately.
     :param cache_output_layer_w_b: Models cache weights and biases for logit computation as NumPy arrays (used with
                                    restrict lexicon).
-    :return: List of models, source vocabulary, target vocabulary.
+    :return: List of models, source vocabulary, target vocabulary, source factor vocabularies.
     """
-    models, source_vocabs, target_vocabs = [], [], []
+    models, source_vocabs, source_factor_vocabs, target_vocabs = [], [], [], []
     if checkpoints is None:
         checkpoints = [None] * len(model_folders)
+
+    source_factor_vocabs = []
     for model_folder, checkpoint in zip(model_folders, checkpoints):
         source_vocabs.append(vocab.vocab_from_json_or_pickle(os.path.join(model_folder, C.VOCAB_SRC_NAME)))
         target_vocabs.append(vocab.vocab_from_json_or_pickle(os.path.join(model_folder, C.VOCAB_TRG_NAME)))
-        model = InferenceModel(model_folder=model_folder,
-                               context=context,
-                               beam_size=beam_size,
-                               batch_size=batch_size,
-                               softmax_temperature=softmax_temperature,
-                               checkpoint=checkpoint,
-                               decoder_return_logit_inputs=decoder_return_logit_inputs,
-                               cache_output_layer_w_b=cache_output_layer_w_b)
-        models.append(model)
+
+        inf_model = InferenceModel(model_folder=model_folder,
+                                   context=context,
+                                   beam_size=beam_size,
+                                   batch_size=batch_size,
+                                   softmax_temperature=softmax_temperature,
+                                   checkpoint=checkpoint,
+                                   decoder_return_logit_inputs=decoder_return_logit_inputs,
+                                   cache_output_layer_w_b=cache_output_layer_w_b)
+        models.append(inf_model)
```

Comment:
good idea with the more explicit variable naming. Lets double down on this and call the list `inference_models` and the individual one `inference_model`.